api/service/syncing/syncing.go

```diff
@@ -761,14 +775,33 @@ Loop:
 			if size > SyncLoopBatchSize {
 				size = SyncLoopBatchSize
 			}
-			err := ss.ProcessStateSync(startHash[:], size, bc, worker)
+
+			retryCount := 0
+			var err error
+			for err == nil && retryCount < stateSyncRetryLimit {
+				err = ss.ProcessStateSync(startHash[:], size, bc, worker)
+				if err != nil {
+					utils.Logger().Warn().Err(err).
+						Msgf("[SYNC] ProcessStateSync failed (isBeacon: %t, ShardID: %d, otherHeight: %d, currentHeight: %d, retryCount: %d)",
+							isBeacon, bc.ShardID(), otherHeight, currentHeight, retryCount)
+					if err != nil {
+						ss.syncConfig.ForEachPeer(func(configPeer *SyncPeerConfig) (brk bool) {
+							ss.syncConfig.failedPeers = append(ss.syncConfig.failedPeers, configPeer)
```

Comment:
should we consider clean failedPeers after a given period of time or blacklist forever?